sbi/samplers/mcmc/pymc_wrapper.py

```diff
@@ -0,0 +1,218 @@
+from typing import Any, Callable, Optional
+
+import numpy as np
+import pymc
+import pytensor.tensor as pt
+import torch
+from arviz.data import InferenceData
+
+from sbi.utils import tensor2numpy
+
+
+class PyMCPotential(pt.Op):  # type: ignore
+    """PyTensor Op wrapping a callable potential function"""
+
+    itypes = [pt.dvector]  # expects a vector of parameter values when called
+    otypes = [
+        pt.dscalar,
+        pt.dvector,
+    ]  # outputs a single scalar value (the potential) and gradients for every input
+    default_output = 0  # return only potential by default
+
+    def __init__(
+        self,
+        potential_fn: Callable,
+        device: str,
+        track_gradients: bool = True,
+    ):
+        """PyTensor Op wrapping a callable potential function for use
+        with PyMC samplers.
+
+        Args:
+            potential_fn: Potential function that returns a potential given parameters
+            device: The device to which to move the parameters before evaluation.
+            track_gradients: Whether to track gradients from potential function
+        """
+        self.potential_fn = potential_fn
+        self.device = device
+        self.track_gradients = track_gradients
+
+    def perform(self, node: Any, inputs: Any, outputs: Any) -> None:
+        """Compute potential and possibly gradients from input parameters
+
+        Args:
+            node: A "node" that represents the computation, handled internally
+                by PyTensor.
+            inputs: A sequence of inputs to the operation of type `itypes`. In this
+                case, the sequence will contain one array containing the
+                simulator parameters.
+            outputs: A sequence allocated for storing operation outputs. In this
+                case, the sequence will contain one scalar for the computed potential
+                and an array containing the gradient of the potential with respect
+                to the simulator parameters.
+        """
+        # unpack and handle inputs
+        params = inputs[0]
+        params = (
+            torch.tensor(params)
+            .to(device=self.device, dtype=torch.float32)
+            .requires_grad_(self.track_gradients)
+        )
+
+        # call the potential function
+        energy = self.potential_fn(params, track_gradients=self.track_gradients)
```

Comment:
note that the `pyro_potential_wrapper` returns the negative potential. You are using that one for pymc too, aren't you?

Reply:
No we aren't using the `pyro_potential_wrapper`, PyMC expects the un-flipped value from what I can tell